system/lib/wasmfs/js_api.cpp

```diff
@@ -151,6 +151,14 @@ int _wasmfs_symlink(char* old_path, char* new_path) {
   return __syscall_symlink((intptr_t)old_path, (intptr_t)new_path);
 }
 
+int _wasmfs_readlink(char* path, char* buf, size_t bufsiz) {
+  int err = __syscall_readlinkat(AT_FDCWD, (intptr_t)path, (intptr_t)buf, bufsiz);
```

Comment:
The error handling should be fixed.

Reply:
Oops, this was left over from before we discussed the error problem